automatic_push/utils.py

```diff
@@ -0,0 +1,166 @@
+# Copyright The IETF Trust 2023, All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains common functions used in the automatic_push directory.
+"""
+
+__author__ = 'Bohdan Konovalenko'
+__copyright__ = 'Copyright The IETF Trust 2023, All Rights Reserved'
+__license__ = 'Apache License, Version 2.0'
+__email__ = 'bohdan.konovalenko@pantheon.tech'
+
+import json
+import logging
+import os
+import sys
+from configparser import ConfigParser
+
+import requests
+from git import GitCommandError
+
+from utility import repoutil
+from utility.staticVariables import github_url
+
+
+def update_forked_repository(yang_models: str, forked_repo_url: str, logger: logging.Logger):
```

Comment:
As this method works with Github repos, cannot we move this functionality to repoutil.py?

Reply:
I believe that this one is ```automatic_push``` directory specific, I moved pushing of all untracked files because we can use it in some other scripts in the future but not sure about these functions that are now in ```utils.py```